$NetBSD: patch-af,v 1.2 2003/10/04 20:22:35 atatat Exp $

Deal with recent procfs and kernfs churn on the -current branch.

--- dialects/n+obsd/dlsof.h.orig	2003-06-11 07:44:54.000000000 -0400
+++ dialects/n+obsd/dlsof.h
@@ -62,6 +62,7 @@
 #define	_KERNEL
 #  if	defined(NETBSDV)
 struct buf;	/* dummy for function prototype in <sys/buf.h> */
+struct uio;	/* dummy for function prototype in <sys/buf.h> */
 #  endif	/* defined(NETBSDV) */
 #include <sys/ucred.h>
 # endif	/* (defined(OPENBSDV) && OPENBSDV>=3030)
@@ -253,6 +254,9 @@ struct sockproto {
 #define	_KERNEL
 #include <miscfs/kernfs/kernfs.h>
 #undef	_KERNEL
+#  if	defined(HASKERNFS_KFS_KT)
+#define kf_kt kfs_kt
+#  endif /* defined(HASKERNFS_KFS_KT) */
 # endif	/* defined(HASKERNFS) */
 
 # if	defined(HASNULLFS)
@@ -266,7 +270,24 @@ struct sockproto {
 # endif	/* defined(HASNULLFS) */
 
 # if	defined(HASPROCFS)
+#  if	defined(HASPROCFS_PFSROOT)
+#define _KERNEL
+#  endif /* defined(HASPROCFS_PFSROOT) */
 #include <miscfs/procfs/procfs.h>
+#  if	defined(HASPROCFS_PFSROOT)
+#undef _KERNEL
+#define Proot PFSroot
+#define Pproc PFSproc
+#define Pcurproc PFScurproc
+#define Pmem PFSmem
+#define Pregs PFSregs
+#define Pfile PFSfile
+#define Pfpregs PFSfpregs
+#define Pctl PFSctl
+#define Pstatus PFSstatus
+#define Pnote PFSnote
+#define Pnotepg PFSnotepg
+#  endif /* defined(HASPROCFS_PFSROOT) */
 #include <machine/reg.h>
 # endif	/* defined(HASPROCFS) */
 
